udata/core/dataset/api.py

```diff
@@ -148,6 +150,15 @@ def parse_filters(datasets, args):
                 pass
             else:
                 datasets = datasets.filter(id__in=[d.id for d in topic.datasets])
+        if args.get("dataservice"):
+            if not ObjectId.is_valid(args["dataservice"]):
+                api.abort(400, "Dataservice arg must be an identifier")
+            try:
+                dataservice = Dataservice.objects.get(id=args["dataservice"])
+            except Dataservice.DoesNotExist:
+                pass
```

Comment:
Should we abort 404 instead of passing?

Reply:
It's copy/paste from the topic logic above. We could change this in another PR if it's not wanted…